### PR TITLE
Add `post_index` signal

### PR DESCRIPTION
**Pull request by saadmk11**
Closes https://github.com/django-es/django-elasticsearch-dsl/issues/298

### CODE REVIEW

**Comment by safwanrahman** [comment]
Thanks @saadmk11 for the PR. It looks very nice with the docuemntation. I will take a closer look into it.

File: django_elasticsearch_dsl/documents.py [hunk 1]
```diff
@@ -143,7 +144,14 @@ def to_field(cls, field_name, model_field):
             )
 
     def bulk(self, actions, **kwargs):
-        return bulk(client=self._get_connection(), actions=actions, **kwargs)
+        response = bulk(client=self._get_connection(), actions=actions, **kwargs)
+        # send post index signal
+        post_index.send(
+            sender=self.__class__,
+            instance=self,
```
Comment (safwanrahman) [inline django_elasticsearch_dsl/documents.py:151]: I think you should pass the actions as well to the signal. Without it, it is quite hard to determine which object is indexed to Elasticsaerch.

**Comment by saadmk11** [comment]
@safwanrahman Thanks for reviewing the PR, I have updated the PR with the suggested changes.

File: tests/test_signals.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        car = Car(
+            pk=51,
+            name="Type 57"
+        )
+
+        doc.update(car)
+
+        bulk.assert_called_once()
+
+        post_index.send.assert_called_once_with(
```
Comment (safwanrahman) [inline tests/test_signals.py:36]: It is better to register a signal function and check whether the function is called with proper values

Comment (saadmk11) [reply tests/test_signals.py]: @safwanrahman Updated the tests.

**Comment by safwanrahman** [comment]
Looks Good to me! r+ 🤩🚀
@saadmk11 you rock! 🥳

**Pull request merged**